packages/ui-causal-graph-editor/src/shared/graph-layout/planar-layout.tsx

```diff
@@ -96,6 +117,18 @@ function customDecross(layers: SugiNode<{ ord?: number }, unknown>[][]): void {
     });
 }
 
+/**
+ * Gets the layering algorithm for a given LayeringAlgorithms enum value
+ *
+ * @param algorithm the layering algorithm to use
+ */
+function getLayeringAlgorithm(algorithm: LayeringAlgorithms): Layering<DagNodeData, any> {
+    if (algorithm === LayeringAlgorithms.LONGEST_PATH) {
+        return layeringLongestPath();
+    }
+    return layeringSimplex();
```

Comment:
I had a `switch` initially, but remembered us talking about this in the past and at that time we said that if there were only two that `if` was considered easier to read. Do you think it should generally be a `switch` for these cases even if there are only two options?

Reply:
It's true this is easier to read but if we were to add another algo to the enum we would be mapping those to simplex in the else clause, whereas if we had a switch I think TypeScript would should at us for not handling the extra case explicitly. Tradeoffs either way, no strong feelings - I'm fine with keeping it as is in this particular scenario, especially since the library is in maintenance so it's unlikely more algos will be added.

Sorry if I'm giving conflicting information on the topic, my opinions also evolve over time 😅

Reply:
Makes sense, thank you :)